ibis/backends/duckdb/tests/test_register.py

```diff
@@ -0,0 +1,67 @@
+import contextlib
+import os
+from pathlib import Path
+
+import pytest
+
+import ibis
+from ibis.backends.conftest import read_tables
+
+
+@contextlib.contextmanager
+def pushd(new_dir):
+    previous_dir = os.getcwd()
+    os.chdir(new_dir)
+    yield
+    os.chdir(previous_dir)
```

Comment:
```suggestion
    try:
        yield
    finally:
        os.chdir(previous_dir)
```

So that you `chdir` back to the original directory even if an exception occurs in the yieldee.